tfhe/src/high_level_api/integers/tests_unsigned.rs

```diff
@@ -779,12 +780,43 @@ fn test_integer_casting() {
 
     // Casting to self, it not useful but is supported
     {
-        let clear = 43_129u16;
         let a = FheUint16::encrypt(clear, &client_key);
         let a = FheUint16::cast_from(a);
         let da: u16 = a.decrypt(&client_key);
         assert_eq!(da, clear);
     }
+
+    // Downcasting to smaller signed integer then Upcasting back to unsigned
+    {
+        let clear = rng.gen_range((i16::MAX) as u16 + 1..u16::MAX);
```

Comment:
not sure I understand why the range is written this way

Reply:
Im downcasting a u16 to an i8, and wanted the u16 value to be out of the range of the i8, I could have use (i8::MAX as u16)+ 1 instead.

not sure doing that range is that useful

Reply:
ah I know what confused me it's the + 1..u16::MAX